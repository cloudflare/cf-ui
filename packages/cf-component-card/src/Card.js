import React from 'react';

import PropTypes from 'prop-types';

class Card extends React.Component {
  render() {
    return (
      <section id={this.props.id} className="cf-card">
        {this.props.children}
      </section>
    );
  }
}

Card.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node
};

export default Card;
