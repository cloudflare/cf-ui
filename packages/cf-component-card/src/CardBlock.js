import React, { PropTypes } from 'react';

class CardBlock extends React.Component {
  render() {
    const className = 'cf-card__block';

    return (
      <div className={className}>
        {this.props.children}
      </div>
    );
  }
}

CardBlock.propTypes = {
  children: PropTypes.node
};

export default CardBlock;
