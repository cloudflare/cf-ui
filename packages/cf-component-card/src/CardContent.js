import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = () => ({
  padding: '1.5rem',

  tablet: {
    display: 'table-cell'
  }
});

const CardContent = ({ className, children }) => (
  <div className={className}>
    {children}
  </div>
);

CardContent.propTypes = {
  children: PropTypes.node
};

export default createComponent(styles, CardContent);
