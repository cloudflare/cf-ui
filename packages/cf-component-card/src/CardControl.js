import React, { PropTypes } from 'react';

class CardControl extends React.Component {
  render() {
    const className = 'cf-card__control' +
      (this.props.wide ? ' cf-card__control--wide' : '');

    return (
      <div className={className}>
        {this.props.children}
      </div>
    );
  }
}

CardControl.propTypes = {
  children: PropTypes.node,
  wide: PropTypes.bool
};

export default CardControl;
