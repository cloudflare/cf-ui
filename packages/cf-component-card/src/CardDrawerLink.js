import React from 'react';
import PropTypes from 'prop-types';
import Link from 'cf-component-link';

class CardDrawerLink extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    e.preventDefault();
    this.props.onClick();
  }

  render() {
    let className = 'cf-card__toolbar_link';

    if (this.props.isActive) {
      className += ' cf-card__toolbar_link--open';
    }

    return (
      <Link
        role="tab"
        id={this.props.id}
        className={className}
        onClick={this.handleClick}
      >
        {this.props.children}
      </Link>
    );
  }
}

CardDrawerLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  id: PropTypes.string,
  children: PropTypes.node
};

export default CardDrawerLink;
