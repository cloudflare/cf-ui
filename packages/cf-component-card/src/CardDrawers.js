import React, { PropTypes } from 'react';
import CardSection from './CardSection';
import CardToolbar from './CardToolbar';
import CardToolbarLink from './CardToolbarLink';
import CardPropTypes from './CardPropTypes';

let UNIQUE_ID = 0;

class CardDrawers extends React.Component {
  constructor(props) {
    super(props);

    this._cardId = UNIQUE_ID++;

    this.state = {
      active: null
    };
  }

  handleLinkClick(id) {
    this.props.onClick && this.props.onClick();

    this.setState(state => ({
      active: state.active === id ? null : id
    }));
  }

  render() {
    const active = this.props.hasOwnProperty('active')
      ? this.props.active
      : this.state.active;

    const links = [];
    const drawers = [];

    this.props.drawers.forEach(drawer => {
      const isActive = drawer.id === active;
      const id = `card-${this._cardId}-drawer-${drawer.id}`;

      links.push(
        <CardToolbarLink
          key={drawer.id}
          id={id}
          isActive={isActive}
          onClick={() => this.handleLinkClick(drawer.id)}
        >
          {drawer.name}
        </CardToolbarLink>
      );

      let className = 'cf-card__drawer';

      if (isActive) {
        className += ' cf-card__drawer--active';
      }

      drawers.push(
        <div
          key={drawer.id}
          role="tabpanel"
          aria-labelledby={id}
          aria-hidden={isActive ? 'false' : 'true'}
          className={className}
        >
          {isActive && drawer.content}
        </div>
      );
    });

    let containerClassName = 'cf-card__drawers_container';

    if (active) {
      containerClassName += ' cf-card__drawers_container--open';
    }

    return (
      <CardSection>
        <CardToolbar controls={this.props.controls} links={links} />
        <div className={containerClassName}>
          {drawers}
        </div>
      </CardSection>
    );
  }
}

CardDrawers.propTypes = {
  onClick: PropTypes.func,

  active: PropTypes.string,
  drawers: CardPropTypes.cardDrawers.isRequired,

  // for an optional control to put on the left side of the toolbar
  controls: PropTypes.any
};

export default CardDrawers;
