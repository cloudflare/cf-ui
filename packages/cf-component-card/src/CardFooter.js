import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = ({ theme }) => ({
  fontSize: theme.fontSize,
  color: theme.color,
  lineHeight: theme.lineHeight,
  marginTop: theme.marginTop
});

const Footer = ({ className, children }) => (
  <div className={className}>{children}</div>
);

Footer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string
};

export default createComponent(styles, Footer);
