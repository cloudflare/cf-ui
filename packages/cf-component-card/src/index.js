import { applyTheme } from 'cf-style-container';

import CardUnstyled from './Card';
import CardTheme from './CardTheme';
import CardBlockUnstyled from './CardBlock';
import CardBlockTheme from './CardBlockTheme';
import CardContentUnstyled from './CardContent';
import CardContentTheme from './CardContentTheme';
import CardControlUnstyled from './CardControl';
import CardControlTheme from './CardControlTheme';
import CardDrawersUnstyled from './CardDrawers';
import CardDrawersTheme from './CardDrawersTheme.js';
import CardMessagesUnstyled from './CardMessages';
import CardMessagesTheme from './CardMessagesTheme';
import CardSectionUnstyled from './CardSection';
import CardSectionTheme from './CardSectionTheme';
import CardTitleUnstyled from './CardTitle';
import CardTitleTheme from './CardTitleTheme';
import CardFooterUnstyled from './CardFooter';
import CardFooterTheme from './CardFooterTheme';
import CardToolbar from './CardToolbar';
import CardToolbarLink from './CardToolbarLink';
import CardPropTypes from './CardPropTypes';

const Card = applyTheme(CardUnstyled, CardTheme);
const CardBlock = applyTheme(CardBlockUnstyled, CardBlockTheme);
const CardContent = applyTheme(CardContentUnstyled, CardContentTheme);
const CardControl = applyTheme(CardControlUnstyled, CardControlTheme);
const CardDrawers = applyTheme(CardDrawersUnstyled, CardDrawersTheme);
const CardMessages = applyTheme(CardMessagesUnstyled, CardMessagesTheme);
const CardSection = applyTheme(CardSectionUnstyled, CardSectionTheme);
const CardTitle = applyTheme(CardTitleUnstyled, CardTitleTheme);
const CardFooter = applyTheme(CardFooterUnstyled, CardFooterTheme);

export {
  Card,
  CardUnstyled,
  CardTheme,
  CardContent,
  CardContentUnstyled,
  CardContentTheme,
  CardControl,
  CardControlUnstyled,
  CardControlTheme,
  CardDrawers,
  CardDrawersUnstyled,
  CardDrawersTheme,
  CardMessages,
  CardMessagesUnstyled,
  CardMessagesTheme,
  CardPropTypes,
  CardSection,
  CardSectionUnstyled,
  CardSectionTheme,
  CardBlock,
  CardBlockUnstyled,
  CardBlockTheme,
  CardTitle,
  CardTitleUnstyled,
  CardTitleTheme,
  CardFooter,
  CardFooterUnstyled,
  CardFooterTheme,
  CardToolbar,
  CardToolbarLink
};
