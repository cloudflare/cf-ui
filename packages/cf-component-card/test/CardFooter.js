import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { CardFooter } from '../../cf-component-card/src/index';

test('should render', () => {
  const snapshot = felaSnapshot(<CardFooter>Card footer</CardFooter>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
