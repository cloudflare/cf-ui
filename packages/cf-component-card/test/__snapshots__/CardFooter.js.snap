// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="a b c d"
>
  Card footer
</div>
`;

exports[`should render 2`] = `
"
.a {
  font-size: 0.8rem
}

.b {
  color: #7c7c7c
}

.c {
  line-height: 1.3
}

.d {
  margin-top: 1.5em
}
"
`;
